sql/src/test/java/io/crate/testing/T3.java

```diff
@@ -22,103 +22,104 @@
-            .add("i", DataTypes.INTEGER)
-            .add("w", DataTypes.LONG)
-            .add("ts_z", DataTypes.TIMESTAMPZ)
-            .add("ts", DataTypes.TIMESTAMP)
-            .build();
-    public static final DocTableRelation TR_5 = new DocTableRelation(T5_INFO);
+    public static final String T1_DEFINITION =
+        "create table doc.t1 (" +
+        "  a text," +
+        "  x int," +
+        "  i int" +
+        ")";
+
+    public static final String T2_DEFINITION =
+        "create table doc.t2 (" +
+        "  b text," +
+        "  y int," +
+        "  i int" +
+        ")";
+
+    public static final String T3_DEFINITION =
+        "create table doc.t3 (" +
+        "  c text," +
+        "  z int" +
+        ")";
+
+    public static final String T4_DEFINITION =
+        "create table doc.t4 (" +
+        "  id int," +
+        "  obj object as (" +
+        "    i int" +
+        "  )," +
+        "  obj_array array(object as (" +
+        "    i int" +
+        "  ))" +
+        ")";
+
+    public static final String T5_DEFINITION =
+        "create table t5 (" +
+        "  i int," +
+        "  w bigint," +
+        "  ts_z timestamp with time zone," +
+        "  ts timestamp without time zone" +
+        ")";
 
     public static final QualifiedName T1 = new QualifiedName(Arrays.asList(Schemas.DOC_SCHEMA_NAME, "t1"));
     public static final QualifiedName T2 = new QualifiedName(Arrays.asList(Schemas.DOC_SCHEMA_NAME, "t2"));
     public static final QualifiedName T3 = new QualifiedName(Arrays.asList(Schemas.DOC_SCHEMA_NAME, "t3"));
     public static final QualifiedName T4 = new QualifiedName(Arrays.asList(Schemas.DOC_SCHEMA_NAME, "t4"));
-    public static final QualifiedName T5 = new QualifiedName(Arrays.asList(Schemas.DOC_SCHEMA_NAME, "t5"));
-
-    public static final ImmutableList<AnalyzedRelation> RELATIONS = ImmutableList.of(TR_1, TR_2, TR_3, TR_4, TR_5);
-    public static final Map<QualifiedName, AnalyzedRelation> SOURCES = ImmutableMap.of(
-        T1, TR_1,
-        T2, TR_2,
-        T3, TR_3,
-        T4, TR_4,
-        T5, TR_5);
+
+    public static final RelationName T1_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t1");
+    public static final RelationName T2_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t2");
+    public static final RelationName T3_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t3");
+    public static final RelationName T4_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t4");
+    public static final RelationName T5_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t5");
+
+    private static final Map<RelationName, String> RELATION_DEFINITIONS = ImmutableMap.of(
+        T1_RN, T1_DEFINITION,
+        T2_RN, T2_DEFINITION,
+        T3_RN, T3_DEFINITION,
+        T4_RN, T4_DEFINITION,
+        T5_RN, T5_DEFINITION);
+
+    public static ImmutableList<AnalyzedRelation> relations(ClusterService clusterService) {
```

Comment:
```suggestion
    public static List<AnalyzedRelation> relations(ClusterService clusterService) {
```

To not expose implementation detail across packages/classes